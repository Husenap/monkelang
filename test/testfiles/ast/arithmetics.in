2+3;
1-2-3;
4-3+2-5+1;
