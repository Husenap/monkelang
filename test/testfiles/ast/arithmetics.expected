PrintStmt: 
  BinOpExpr: +
    NumberExpr: 1
    NumberExpr: 2
PrintStmt: 
  BinOpExpr: +
    BinOpExpr: -
      NumberExpr: 1
      NumberExpr: 2
    NumberExpr: 3
ExprStmt: 
  BinOpExpr: ==
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: !=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: <
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: <=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: >
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: >=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: -
    BinOpExpr: +
      NumberExpr: 2
      NumberExpr: 3
    NumberExpr: 4
ExprStmt: 
  BinOpExpr: +
    NumberExpr: 1
    BinOpExpr: *
      NumberExpr: 2
      NumberExpr: 3
ExprStmt: 
  BinOpExpr: ==
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: !=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: <
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: <=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: >
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: >=
    NumberExpr: 2
    NumberExpr: 2
ExprStmt: 
  BinOpExpr: -
    BinOpExpr: +
      NumberExpr: 2
      NumberExpr: 3
    NumberExpr: 4
ExprStmt: 
  BinOpExpr: +
    NumberExpr: 1
    BinOpExpr: *
      NumberExpr: 2
      NumberExpr: 3
